error-issue/src/service_error_monitor.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2020 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ErrorMonitor} from './error_monitor';
+import {ServiceGroup, Stackdriver} from 'error-issue-bot';
+import {StackdriverApi} from './stackdriver_api';
+
+export enum ServiceName {
+  PRODUCTION = 'CDN Production',
+  DEVELOPMENT = 'CDN Development',
+  EXPERIMENTS = 'CDN Experiments',
+  NIGHTLY = 'CDN Nightly',
+}
+
+/**
+ * Set of service groups to monitor and details for scaling frequency.
+ * Note: These values do not need to be exact; the order-of-magnitude is what's
+ * important here.
+ */
+const SERVICE_GROUPS: Record<ServiceName, ServiceGroup> = {
+  'CDN Production': {diversionPercent: 0.98, throttleRate: 0.1},
+  'CDN Development': {diversionPercent: 0.01, throttleRate: 1},
+  'CDN Experiments': {diversionPercent: 0.01, throttleRate: 1},
```

Comment:
Just so I understand, do both the 1% and opt-in channels fall in these groups?

Reply:
Yes; for the purposes of error-reporting, opt-in is (and was previously) treated the same as 1%. A useful reference here (and I can include a link in the comment, if you think it wise) is the [error-tracker code](https://github.com/ampproject/error-tracker/blob/master/utils/rtv/release-channels.js) that maps from RTV prefixes to a service group and version name. Replicating below for convenience:

```javascript
module.exports = {
  '00': { group: 'Development', name: 'Experimental' },
  '01': { group: 'Production', name: 'Stable' },
  '02': { group: 'Production', name: 'Control' },
  '03': { group: 'Development', name: 'Beta' },
  '04': { group: 'Nightly', name: 'Nightly' },
  '05': { group: 'Nightly', name: 'Nightly-Control' },
  '10': { group: 'Experiments', name: 'Experiment-A' },
  '11': { group: 'Experiments', name: 'Experiment-B' },
  '12': { group: 'Experiments', name: 'Experiment-C' },
  // Ads error reporting will get all of the below channels, so the service
  // bucket names can be more verbose.
  '20': { group: 'Inabox-Control-A', name: 'Inabox-Control-A' },
  '21': { group: 'Inabox-Experiment-A', name: 'Inabox-Experiment-A' },
  '22': { group: 'Inabox-Control-B', name: 'Inabox-Control-B' },
  '23': { group: 'Inabox-Experiment-B', name: 'Inabox-Experiment-B' },
  '24': { group: 'Inabox-Control-C', name: 'Inabox-Control-C' },
  '25': { group: 'Inabox-Experiment-C', name: 'Inabox-Experiment-C' },
};
```

Reply:
Ahh I see, thanks!